### PR TITLE
Use selective top-level module imports in std.{base64,csv,datetime}

### PR DESCRIPTION
**Pull request by wilzbach**
Doing this in multiple parts.

See also: https://github.com/dlang/phobos/pull/7024

### CODE REVIEW

File: std/base64.d [hunk 1]
```diff
@@ -1764,6 +1765,7 @@ class Base64Exception : Exception
     import std.algorithm.comparison : equal;
     import std.algorithm.sorting : sort;
     import std.conv;
+    import std.exception : assertThrown;
```
Comment (thewilsonator) [inline std/base64.d]: the comment said this was for enforce...

Comment (thewilsonator) [reply std/base64.d]: Oh this is in a unittest. whoops.

Comment (wilzbach) [reply std/base64.d]: Yeah, but the comment is old and wasn't enforced by the compiler.

**Comment by thewilsonator** [comment]
```
std/datetime/timezone.d(1409): Error: template instance `isOutputRange!(W, char)` template `isOutputRange` is not defined
std/datetime/timezone.d(1403): Error: none of the overloads of `toISOString` are callable using argument types `(Appender!string, Duration)`, candidates are:
std/datetime/timezone.d(1398):        `std.datetime.timezone.SimpleTimeZone.toISOString(Duration utcOffset)`
std/datetime/timezone.d(1408):        `std.datetime.timezone.SimpleTimeZone.toISOString(W)(ref W writer, Duration utcOffset) if (isOutputRange!(W, char))`
```

**Pull request merged**